src/sentry/static/sentry/app/components/actions/actionLink.tsx

```diff
@@ -1,79 +1,66 @@
 import React from 'react';
 import styled from '@emotion/styled';
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 
-import Confirm from 'app/components/confirm';
+import ActionButton from './button';
+import ConfirmableAction from './confirmableAction';
 
-type DefaultProps = {
-  disabled: boolean;
-  shouldConfirm: boolean;
-};
+type ConfirmableActionProps = React.ComponentProps<typeof ConfirmableAction>;
 
-type ActionLinkProps = DefaultProps & {
+type Props = Omit<ConfirmableActionProps, 'onConfirm' | 'confirmText' | 'children'> & {
   title: string;
-  message?: React.ReactNode;
+  children?: React.ReactNode;
   onAction: () => void;
+  type?: 'button';
+  disabled?: boolean;
+  className?: string;
+  shouldConfirm?: boolean;
   confirmLabel?: string;
-} & React.HTMLAttributes<HTMLDivElement>;
+} & Partial<React.ComponentProps<typeof ActionButton>>;
 
-export default class ActionLink extends React.Component<ActionLinkProps> {
-  static propTypes = {
-    className: PropTypes.any,
-    title: PropTypes.string,
-    message: PropTypes.node,
-    disabled: PropTypes.bool,
-    onAction: PropTypes.func.isRequired,
-    shouldConfirm: PropTypes.bool,
-    confirmLabel: PropTypes.string,
-  };
-
-  static defaultProps: DefaultProps = {
-    shouldConfirm: false,
-    disabled: false,
-  };
-
-  render() {
-    const {
-      shouldConfirm,
-      message,
-      className,
-      title,
-      onAction,
-      confirmLabel,
-      disabled,
-      children,
-    } = this.props;
-    const testId = title
-      ? 'action-link-' + title.toLowerCase().replace(/ /g, '-')
-      : 'action-link';
-
-    if (shouldConfirm && !disabled) {
-      return (
-        <Confirm message={message} confirmText={confirmLabel} onConfirm={onAction}>
-          <a className={className} title={title} aria-label={title}>
-            {' '}
-            {children}
-          </a>
-        </Confirm>
-      );
-    } else {
-      return (
-        <ActionLinkAnchor
-          data-test-id={testId}
-          aria-label={title}
-          className={classNames(className, {disabled})}
-          onClick={disabled ? undefined : onAction}
-          disabled={disabled}
-        >
-          {children}
-        </ActionLinkAnchor>
-      );
-    }
-  }
+export default function ActionLink({
+  message,
+  className,
+  title,
+  onAction,
+  type,
+  confirmLabel,
+  disabled,
+  children,
+  shouldConfirm,
+  ...props
+}: Props) {
+  return (
+    <ConfirmableAction
+      shouldConfirm={shouldConfirm}
+      disabled={disabled}
+      message={message}
+      confirmText={confirmLabel}
+      onConfirm={onAction}
+    >
+      <ActionLinkAnchor
+        as={type === 'button' ? ActionButton : 'a'}
```

Comment:
this component is an anchor element, so this would not work, should we do something like:

`type === 'button' ? <ActionButton {...props} /> : <ActionLinkAnchor {...props}  />`

Reply:
I'd prefer to leave it as we want the `ActionLinkAnchor` styles to also apply to `ActionButton`, but I renamed `ActionLinkAnchor` to be more generic to reflect this